generator.go

```diff
@@ -336,5 +341,8 @@ func fieldType(f *fieldValues) string {
 	if f.IsRepeated {
 		return t + "[]"
 	}
+	if f.IsOptional {
+		t = fmt.Sprintf("%s | null", t)
```

Comment:
in general even scalars should be optional in a sense that there should be a default value if they are not specified.
Though this is a good first steps for messages - where the default value is null